modules/core/04-channel/keeper/packet.go

```diff
@@ -128,14 +128,14 @@ func (k Keeper) RecvPacket(
 		return errorsmod.Wrap(types.ErrChannelNotFound, packet.GetDestChannel())
 	}
 
-	if !collections.Contains(channel.State, []types.State{types.OPEN, types.TRYUPGRADE, types.ACKUPGRADE}) {
-		return errorsmod.Wrapf(types.ErrInvalidChannelState, "expected channel state to be one of [%s, %s, %s], but got %s", types.OPEN.String(), types.TRYUPGRADE.String(), types.ACKUPGRADE.String(), channel.State.String())
+	if !collections.Contains(channel.State, []types.State{types.OPEN, types.STATE_FLUSHING}) {
+		return errorsmod.Wrapf(types.ErrInvalidChannelState, "expected channel state to be one of [%s, %s], but got %s", types.OPEN, types.STATE_FLUSHING, channel.State)
 	}
 
-	// in the case of the channel being in TRYUPGRADE or ACKUPGRADE we need to ensure that the channel is not in flushing,
-	// and that the counterparty last sequence send is less than or equal to the packet sequence.
+	// in the case of the channel being in FLUSHING we need to ensure that the the counterparty last sequence send
+	// is less than or equal to the packet sequence.
 	if counterpartyLastSequenceSend, found := k.GetCounterpartyLastPacketSequence(ctx, packet.GetDestPort(), packet.GetDestChannel()); found {
-		if channel.FlushStatus != types.FLUSHING || packet.GetSequence() > counterpartyLastSequenceSend {
+		if channel.State != types.STATE_FLUSHING || packet.GetSequence() > counterpartyLastSequenceSend {
 			return errorsmod.Wrapf(types.ErrInvalidFlushStatus, "expected channel flush status to be (%s) when counterparty last sequence send (%d) is set, failed to recv packet (%d)", types.FLUSHING, counterpartyLastSequenceSend, packet.GetSequence())
```

Comment:
We'll no longer be storing the counterparty last packet sequence send under its own key in state as we now store the entire counterparty upgrade.

I think this code will need to be replaced with something like this:

```golang
if channel.State == types.STATE_FLUSHING {
	counterpartyUpgrade, found := k.GetCounterpartyUpgrade(ctx, packet.GetDestPort(), packet.GetDestChannel())
	if !found {
		return errorsmod.Wrapf(types.ErrUpgradeNotFound, "counterparty upgrade not found for channel: %s", packet.GetDestChannel())
	}

	if packet.GetSequence() > counterpartyUpgrade.LatestSequenceSend {
		return errorsmod.Wrapf(
			types.ErrInvalidPacket,
			"failed to receive packet (%d), cannot flush packet at sequence greater than counterparty last sequence send (%d)", packet.GetSequence(), counterpartyUpgrade.LatestSequenceSend,
		)
	}
}
```

Reply:
Yeah! So now that we will store the full counterparty upgrade we no longer need to set the counterparty last sequence send under its own key. 

There is a catch here but I don't see it being a big concern really. Basically, the counterparty timeout and last packet send will only be fulfilled in ack or confirm handshake steps. So you might have a relayer attempt a `recvPacket` after the try handler, when the channel is in `FLUSHING` but an error is returned as `packet.GetSequence() > 0` - zero being the `counterpartyUpgrade.LastSequenceSend` as it hasn't been set yet. 

But, once the relayer does the next step of the upgrade handshake and sets the value, then the `recvPacket` should be able to be retried successfully.

Reply:
I think I recall just advising to punt removing the state key and getter/setter for counterparty last send sequence until we had some more progress. I think we are probably very close to that point now

Reply:
ah, let me ping ya on the point you raised in slack